kythe/docs/schema/modules.txt

```diff
@@ -0,0 +1,163 @@
+components are to be *imported* from other modules. It may also be possible
+to refer to the module itself as a first-class value in a program.
+
+In the following sections, we'll discuss common user expectations for
+cross-references in programs that use modules. The representation of a module
+itself is not in this document's scope.
+
+== Simple imports
+
+The second line makes the module named `fmt` available in scope as `fmt`. The
+exact semantics of what the local `fmt` is can differ depending on the
+language. In some languages, `fmt` is simply a local variable that points to
+the (first-class) module object; in others, it is part of a special category of
+identifiers that cannot be rebound. Users generally prefer to see any use of
+`fmt` *ref* the node representing the module's definition instead of the
+definition of the local alias, which may not even have a sensible syntactic
+location to anchor to:
+
+[source,go]
+----
+import (
+//- @"\"fmt\"" ref/imports FmtModule
+  "fmt"
+)
+//- @fmt ref FmtModule
+fmt.Sprint
+----
+
+With these edges, users who attempt to navigate from the module's use site will
+be brought to the definition of `fmt`, not to the definition of the local
+alias. In the same way, users investigating cross-references from the module's
+definition will see every use of that module in the database (up to re-exporting
+and vexing control-flow), not just the places where it's been imported. In
+addition, other information (like documentation) will be available at every use
+site.
+
+Some languages allow particular values to be imported into local scope. These
+should be handled in the same way as module definitions:
+
+[source,javascript]
+----
+//- @value ref/imports Value
+//- @"'./module'" ref/imports Module
+import {value} from './module';
+//- @value ref Value
+value;
+----
+
+== Imports with renaming
+
+It's sometimes possible (or is even required) to choose a different name for the
+local binding created by an import from the target definition's. In this case,
+the local name *does* have a sensible syntactic location and is semantically
+relevant to the programmer. A use site of the local name refers to both the
+module (or value) being imported and to the local alias:
+
+[source,javascript]
+----
+//- @"'./module'" ref/imports Module
+//- @defaultExport defines/binding DefaultExport
+//- @defaultExport ref/imports DefaultExportDef
+//- DefaultExport.subkind import
+import defaultExport from './module';
+//- @"'./module'" ref/imports Module
+//- @mod_imp defines/binding ModImp
+//- @mod_imp ref/imports Module
+//- ModImp.subkind import
+import * as mod_imp from './module';
+//- @"'./module'" ref/imports Module
+//- @value ref/imports Value
+//- @renamedValue defines/binding RenamedValue
+//- RenamedValue.subkind import
+import {value as renamedValue} from './module';
+//- @defaultExport ref DefaultExportDef
+//- @defaultExport ref DefaultExport
```

Comment:
My recollection was that we'd agreed not to emit duplicate `ref` nodes, but using alias at the import site to accomplish this.

Reply:
(merging this thread with the `aliases` suggestion; I need to check that this has the behavior we want wrt clustering)

Reply:
My recollection and meeting notes suggest that the behavior we want is:

1) Clicking on a reference in a file should jump to
  a) The import line, if renamed
  b) The exported entity, if not
2) Clicking on a definition should show all uses, seeing through the renaming

I believe we'd settled on using a combination of aliases, clustering and `target_definition` to accomplish those, such that indexers would:

1) emit `ref/imports` at the import site
2) {name} -- defines/binding -> {alias} -- aliases -> {original} for renames (at least, I think doing this for not-renames is also fine when combined with the rest of the changes)
3) clustering would include follow reverse alias edges (and that accurate counts don't matter)
4) 1.b from above would be accomplished by using the imported node as the `target_definition` in `DecorationsReply` without necessarily changing the schema.